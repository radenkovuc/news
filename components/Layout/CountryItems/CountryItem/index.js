import styled from 'styled-components'
import React from 'react'
import withTranslation from '../../../HOCs/withTranslation'

const Item = styled.div`
  display: flex;
  border-left: 1px solid #444;
  justify-content: center;
  background: ${(props) => (props.isDisabled ? '#FF6347' : props.isActive ? '#777777' : '#ffffff')};
  color: ${(props) => (props.isActive ? '#ffffff' : '#000000')};
  cursor: ${(props) => (props.isDisabled ? 'default' : 'pointer')};
  flex: 1;

  @media only screen and (min-width: 600px) {
    flex: none;
  }
`
const Text = styled.span`
  padding: 20px;
  font-size: 20px;
  font-family: 'Nunito Sans', sans-serif;
`

type Props = {
  text: String,
  isActive: Boolean,
  isDisabled: Boolean,
  onClick: Boolean,
  t: Function
}

const CountryItem = (props: Props) => {
  const {text, isActive, isDisabled, onClick, t} = props

  return (
    <Item isActive={isActive} isDisabled={isDisabled} onClick={onClick}>
      <Text>{t(text)}</Text>
    </Item>
  )
}

CountryItem.defaultProps = {
  text: '',
  expanded: false,
  isDisabled: false,
  onClick: () => {},
  t: (t) => t
}

export default withTranslation(CountryItem)
