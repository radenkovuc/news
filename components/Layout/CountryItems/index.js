import styled from 'styled-components'
import React from 'react'
import CountryItem from './CountryItem'
import withContext from '../../HOCs/withContext'
import {COUNTRIES} from '../../../common/consts.json'

const Wrapper = styled.div`
  display: flex;
  align-self: center;
  margin: 0;
  width: 100%;
  border-bottom: 1px solid #444;

  @media only screen and (min-width: 600px) {
    width: auto;
  }
`

const CountryItems = (props) => {
  const {
    isDisabled,
    appContext: {selectedCountry, setContextData}
  } = props

  const countries = []

  // eslint-disable-next-line no-unused-vars
  for (let key in COUNTRIES) {
    // eslint-disable-next-line no-prototype-builtins
    if (COUNTRIES.hasOwnProperty(key)) {
      const {langKey} = COUNTRIES[key]
      countries.push(
        <CountryItem
          key={key}
          text={langKey}
          isActive={selectedCountry === key}
          isDisabled={isDisabled}
          onClick={() => !isDisabled && setContextData({selectedCountry: key})}
        />
      )
    }
  }

  return <Wrapper>{countries}</Wrapper>
}

export default withContext(CountryItems)
