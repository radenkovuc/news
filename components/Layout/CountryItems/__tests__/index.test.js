import React from 'react'
import renderer from 'react-test-renderer'

import {createTextComponent} from '../../../../test/TestUtil'

import CountryItems from '../'

describe('<CountryItems />', () => {
  it('match snapshot', () => {
    const tree = renderer.create(createTextComponent(<CountryItems/>)).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
