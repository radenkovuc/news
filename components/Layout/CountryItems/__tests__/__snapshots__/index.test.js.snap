// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryItems /> match snapshot 1`] = `
<div
  className="Wrapper-sc-176nf4l-0 hSHBvl"
>
  <div
    className="Item-fwh4iu-0 bNiPnD"
    onClick={[Function]}
  >
    <span
      className="Text-fwh4iu-1 KCUNf"
    >
      GB
    </span>
  </div>
  <div
    className="Item-fwh4iu-0 bNiPnD"
    onClick={[Function]}
  >
    <span
      className="Text-fwh4iu-1 KCUNf"
    >
      US
    </span>
  </div>
</div>
`;
